sw/host/ot_certs/src/x509.rs

```diff
@@ -75,6 +76,35 @@ fn asn1octets_to_vec(s: &Asn1OctetStringRef) -> Value<Vec<u8>> {
     Value::literal(s.as_slice().to_vec())
 }
 
+fn asn1time_to_string(time: &Asn1TimeRef) -> Result<Value<String>> {
+    // SAFETY: OpenSSL guarantees that an ASN1_TIME is in fact just a typedef for ASN1_STRING
+    // https://www.openssl.org/docs/man1.1.1/man3/ASN1_TIME_to_generalizedtime.html
+    let time_str = time.as_ptr() as *mut openssl_sys::ASN1_STRING;
+    // SAFETY: the above pointer is guaranteed to be valid since `time` is valid reference.
+    let time_type = unsafe { openssl_sys::ASN1_STRING_type(time_str) };
+    // FIXME: we only support GeneralizedTime and openssl_sys does not export the OpenSSL method to
+    // convert to a GeneralizedTime so we just error out.
+    if time_type == openssl_sys::V_ASN1_UTCTIME {
+        bail!("time uses UtcTime but only GeneralizedTime is supported")
+    }
+    if time_type != openssl_sys::V_ASN1_GENERALIZEDTIME {
+        bail!("time uses type {time_type} but only GeneralizedTime is supported")
+    }
+    // SAFETY: the above pointer is guaranteed to be valid since `time` is valid reference.
+    // From the OpenSSL documentation: ASN1_STRING_get0_data() returns an internal pointer to
+    // the data of x. Since this is an internal pointer it should not be freed or modified in any way.
+    let time_str = unsafe {
+        let len = openssl_sys::ASN1_STRING_length(time_str);
+        let data = openssl_sys::ASN1_STRING_get0_data(time_str);
+        std::slice::from_raw_parts(data, len as usize)
+    };
```

Comment:
How about just use
```rust
let time_str = unsafe { Asn1StringRef::from_ptr(time_str) }.as_str();
```

Reply:
I should I thought about that ^^ Thanks for the suggestion.